servlet/src/main/scala/org/http4s/servlet/ServletIo.scala

```diff
@@ -19,51 +19,72 @@ package servlet
 
 import cats.effect._
 import cats.effect.std.Dispatcher
-import cats.effect.std.Queue
 import cats.syntax.all._
 import fs2._
+import org.http4s.internal.bug
 import org.log4s.getLogger
 
-import java.util.Arrays
+import java.util.concurrent.CancellationException
+import java.util.concurrent.atomic.AtomicReference
 import javax.servlet.ReadListener
 import javax.servlet.WriteListener
 import javax.servlet.http.HttpServletRequest
 import javax.servlet.http.HttpServletResponse
+import scala.annotation.tailrec
 
 /** Determines the mode of I/O used for reading request bodies and writing response bodies.
   */
-sealed trait ServletIo[F[_]] {
+sealed abstract class ServletIo[F[_]: Async] {
```

Comment:
I think if you keep the old definition of just this class, some of the above comments fall into place.

Reply:
Reset to `trait` updating soon